### PR TITLE
Implemented streamed images via SHCreateMemStream

### PR DESCRIPTION
**Pull request by Kolsky**
Reverted changes from the previous merge.
Somewhat mad at myself.

### CODE REVIEW

File: native-windows-gui/src/win32/image_decoder.rs [hunk 1]
```diff
@@ -50,17 +50,29 @@ pub unsafe fn create_decoder_from_file<'a>(fact: &IWICImagingFactory, path: &'a
...
-    //Stream data must never be removed while stream is in use. See https://docs.microsoft.com/en-us/windows/win32/api/wincodec/nf-wincodec-iwicstream-initializefrommemory
-    let r = (&*stream).InitializeFromMemory(stream_data.as_mut_ptr(), stream_data.len() as _);
+    let length = stream_data.len().try_into()
+        .map_err(|_| NwgError::resource_create("Failed to create memory stream from stream data due to length overflow, which must fit in u32"))?;
+    let mem_stream = SHCreateMemStream(stream_data.as_ptr(), length);
+    if mem_stream.is_null() {
+        return Err(NwgError::resource_create("Failed to create memory stream"));
+    }
+
+    let r = (&*stream).InitializeFromIStream(mem_stream);
```
Comment (sopium) [inline native-windows-gui/src/win32/image_decoder.rs:75]: You can pass the `*IStream` returned by `SHCreateMemStream` to `CreateDecoderFromStream` directly😀.

Comment (Kolsky) [reply native-windows-gui/src/win32/image_decoder.rs:75]: Are github users able to commit at the same pull request?

Comment (Kolsky) [reply native-windows-gui/src/win32/image_decoder.rs:75]: Are github users able to commit at the same pull request? @sopium

Comment (Kolsky) [reply native-windows-gui/src/win32/image_decoder.rs:75]: Aight, I'll just copy your changes, as everything works but comments 'n stuff are better

**Pull request merged**